### PR TITLE
EW-10515: add tests that increase stream coverage taken from other runtimes

### PR DESCRIPTION
**Pull request by anonrig**
Related to ticket: [EW-10515](https://jira.cfdata.org/browse/EW-10515)

### CODE REVIEW

File: src/workerd/api/node/tests/module-require-mutable-exports-test.js [hunk 1]
```diff
@@ -48,9 +48,8 @@ export const testPatchingDoesNotAffectNamedExports = {
     timersPromises.setTimeout = patchedSetTimeout;
 
     // The named export from the module namespace should be unaffected
-    const { setTimeout: namedSetTimeout } = await import(
-      'node:timers/promises'
-    );
+    const { setTimeout: namedSetTimeout } =
```
Comment (npaun) [inline src/workerd/api/node/tests/module-require-mutable-exports-test.js:51]: Spurious formatting change?

Comment (anonrig) [reply src/workerd/api/node/tests/module-require-mutable-exports-test.js:51]: This was fixed with https://github.com/cloudflare/workerd/commit/e51caaa81a8960db666eef5a2e982bc285e76706, no need to change this pull-request since there won't be any conflict

Comment (anonrig) [reply src/workerd/api/node/tests/module-require-mutable-exports-test.js:51]: Rebased using Github UI to remove this change.

File: src/workerd/api/tests/compression-streams-edge-cases-test.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    // Verify empty result
+    const dsReader = ds.readable.getReader();
+    const result = await dsReader.read();
+    ok(result.done);
+  },
+};
+
+// Test CompressionStream piped to IdentityTransformStream
+// Inspired by: workerd compression-streams-test.js
+export const compressionPipeToIdentity = {
```
Comment (npaun) [inline src/workerd/api/tests/compression-streams-edge-cases-test.js]: I can't decide if it's redundant but it's pretty close to https://github.com/cloudflare/workerd/blob/637f37fae5660089309263e67c33c077ea3f1ad2/src/workerd/api/tests/compression-streams-test.js#L299

Comment (anonrig) [reply src/workerd/api/tests/compression-streams-edge-cases-test.js]: One of them uses gzip, one of them uses deflate but according to Claude here's the full differences


| Aspect | requestBodyPipethrough | compressionPipeToIdentity |
|--------|--------------------------|----------------------------|
| Compression algorithm | gzip | deflate |
| Data source | Fetches from a service binding (env.SERVICE) | Creates a local ReadableStream |
| What's tested | Decompressing a response body from a fetch | Round-trip: compress → identity transform → decompress |
| IdentityTransformStream | Not used | Core part of the test - data passes through it |
| Dependencies | Requires service binding | Self-contained |
| Expected output | 'hello world '.repeat(100) | 'Piping compression to identity transform.' |

File: src/workerd/api/tests/compression-streams-edge-cases-test.js [hunk 3]
```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2026 Cloudflare, Inc.
```
Comment (jasnell) [inline src/workerd/api/tests/compression-streams-edge-cases-test.js]: Would prefer this moved to the existing compression-streams-test.js to keep them together

File: src/workerd/api/tests/streams-backpressure-test.js [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+    const result1 = await reader.read();
+    strictEqual(result1.value, 1);
+    strictEqual(result1.done, false);
+
+    const result2 = await reader.read();
+    strictEqual(result2.value, 2);
+    strictEqual(result2.done, false);
+
+    ok(pullCount >= 2, 'pull should be called multiple times');
```
Comment (jasnell) [inline src/workerd/api/tests/streams-backpressure-test.js]: This is redundant. You're enqueing pullCount and checking result of the read above and then checking pullCount again here.

File: src/workerd/api/tests/streams-backpressure-test.js [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+    const w2 = writer.write('b');
+    strictEqual(writer.desiredSize, 0);
+
+    const initialReady = writer.ready;
+
+    const w3 = writer.write('c');
+    strictEqual(writer.desiredSize, -1);
+
+    ok(
+      writer.ready !== initialReady || writer.desiredSize <= 0,
```
Comment (jasnell) [inline src/workerd/api/tests/streams-backpressure-test.js:226]: A comment here explaining why the || is necessary would be helpful for others coming into this test later. It's non-obvious.

File: src/workerd/api/tests/streams-backpressure-test.js [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+      'backpressure signal'
+    );
+
+    await Promise.all([w1, w2, w3]);
```
Comment (jasnell) [inline src/workerd/api/tests/streams-backpressure-test.js]: Might be worth using allSettled for these and checking each result. Otherwise if one of the promises rejects it's a bit more difficult to determine *which* one rejected.

**Review by jasnell — APPROVED** [review]
